### PR TITLE
feat(lnd): handle delayed macaroon creation

### PR DESCRIPTION
**Pull request by sangaman**
This improves the initialization flow for LndClient when creating a new wallet in lnd. Sometimes `admin.macaroon` is not created immediately after calling `InitWallet` on lnd. Previously, this could lead to us being stuck in a loop on `awaitWalletInit` immediately after creating or restoring a wallet in lnd. If there are no macaroons by the first time we attempt to verify the connection, we treated it as if lnd had never received an `InitWallet` command and wait for that command to happen, however that call never comes since it had already happened.

Instead, this adds new logic to detect the above scenario and skip ahead to simply wait for `admin.macaroon` to be written to disk and made available for lnd calls by xud.

### CODE REVIEW

**Comment by kilrau** [comment]
This one could potentially fix the xud restart we are currently enforcing in the xud-docker create flow. Let's test without restart with this branch @raladev

File: lib/lndclient/LndClient.ts [hunk 1]
```diff
@@ -384,16 +384,27 @@ class LndClient extends SwapClient {
    * to verify the connection to lnd.
    */
   private awaitWalletInit = async () => {
-    // we are waiting for lnd to be initialized by xud and for the lnd macaroons to be created
-    this.logger.info('waiting for wallet to be initialized...');
-
     /**
-     * A promise that resolves to `true` when the lnd wallet is created via an InitWallet call,
-     * resolves to `false` if we close the client before the lnd wallet is created.
+     * Whether he lnd wallet has been created via an InitWallet call,
```
Comment (erkarl) [inline lib/lndclient/LndClient.ts:388]:
```suggestion
     * Whether the lnd wallet has been created via an InitWallet call,
```

**Comment by sangaman** [comment]
I'm thinking here it might make sense to explicitly wait for admin.macaroon to be created after xud performs the InitWallet call to lnd. That way we can wait to respond to the xud `InitNode` call until all lnds have finished creating admin.macaroon and are ready to receive calls. I'll add that in a separate commit.

**Comment by sangaman** [comment]
Since this appears to fix our restore issues and is working well in manual tests, I'm going to merge this PR as is (with karl's typo fix) and then open a separate PR for my proposal above.

**Comment by kilrau** [comment]
Sounds good. Go ahead, merging is not blocked anymore.

**Pull request merged**